### PR TITLE
improve ExponentialReconnectionPolicy,now can custom max attempts time

### PR DESCRIPTION
**Pull request by vipjml**

### CODE REVIEW

File: cassandra/policies.py [hunk 1]
```diff
@@ -528,7 +528,7 @@ class ExponentialReconnectionPolicy(ReconnectionPolicy):
     a set maximum delay.
     """
 
-    def __init__(self, base_delay, max_delay):
+    def __init__(self, base_delay, max_delay,max_attempts=64):
```
Comment (aholmberg) [inline cassandra/policies.py]: Please use PEP-8 spacing throughout.

File: cassandra/policies.py [hunk 2]
```diff
@@ -528,7 +528,7 @@ class ExponentialReconnectionPolicy(ReconnectionPolicy):
...
         """
         `base_delay` and `max_delay` should be in floating point units of
         seconds.
```
Comment (aholmberg) [inline cassandra/policies.py]: Docstring explanation for max_attempts, similar to ```ConstanctReconnectionPolicy```

**Comment by aholmberg** [comment]
Are you planning to add unit tests?

**Comment by vipjml** [comment]
I add a unit test,fix spacing and docstring

File: cassandra/policies.py [hunk 3]
```diff
@@ -528,10 +528,14 @@ class ExponentialReconnectionPolicy(ReconnectionPolicy):
     a set maximum delay.
     """
 
-    def __init__(self, base_delay, max_delay):
+    def __init__(self, base_delay, max_delay, max_attempts=None):
```
Comment (aholmberg) [inline cassandra/policies.py]: I think we need to keep the default 64 to match the previous behavior if the parameter is unspecified.

Does it ever make sense to try forever (None)? With exponential backoff, the retry interval gets very large, and might lead to situations where reconnects are not very responsive (with a retry scheduled, but *way* in the future).

As an aside, are you trying to solve a specific problem here?

**Comment by aholmberg** [comment]
If you could, please rebase/squash so we don't have so many commits for one change. Alternatively, I can just apply a single patch in your name.

**Comment by vipjml** [comment]
The reason why I want to change max_attempts=None is that, In some situation, the client use up all retries of all nodes, for example,cassandra down for hours,the client will never recover except you recreate a  cassandra.cluster.Cluster or restart the client program. A simple way to regenerate this: create a client with ExponentialReconnectionPolicy(1,2), close cassandra after client connected, then wait for a while, after the client use up all retries, start cassandra, but the client will never recover,keep throwing NoHostAvailable.

**Comment by vipjml** [comment]
I change default max_attempts to 64, but I think default max_attempts-64 should be better for most users

**Comment by aholmberg** [comment]
@vipjml I understand the point and I agree. All I'm trying to do is avoid changing default behavior in a minor version. I can add a TODO there to update on the next major rev.

**Comment by vipjml** [comment]
that is well

**Comment by vipjml** [comment]
this is well !

**Comment by aholmberg** [comment]
Thanks for the contribution!

**Pull request merged**